astropy/stats/setup_package.py

```diff
@@ -2,7 +2,7 @@
 
 import os
 
-import numpy
+import numpy as np
```

Comment:
I guess this doesn't strictly hurt, but numpy is really not used to access anything from it here, but rather as the package, to get its include files. By using the full name, it is immediately clear that this is not the usual situation, so this change makes the code less clear, not more.

Below I suggest `from numpy import get_include as get_numpy_include`, but I feel like, again, we are coding around style enforcers, which seems incredibly silly.

Reply:
There's no need to clutter `pyproject.toml` with this.

Reply:
No, we are making all customisations in `.ruff.toml`, right?

Reply:
`.ruff.toml` is meant to be a temporary file and the long-term goal is to get rid of it entirely. All the permanent configuration is in `pyproject.toml`.

Reply:
Hmm, there are already some 70 per-file and per-subpackage exclusions in there – are those all meant to be temporary? 😳

Reply:
For the most part yes. It is possible that we might want to ignore a small number of rules currently listed in `.ruff.toml` permanently, but whenever we make such a decision the relevant ignores should be moved to `pyproject.toml`. If there is some rule we usually want to enforce then it is best to use the `noqa` instructions for the exceptions.

Reply:
I'd certainly find it preferable then to "clutter" `pyproject.toml` with one exception line than to add `noqa`s to every single `setup_package.py`.

Reply:
Most importantly [Ruff rule RUF100](https://docs.astral.sh/ruff/rules/unused-noqa/) checks for unnecessary `noqa` instructions, so if anything ever changes in Ruff itself or in our configuration then our code will not be littered with obsolete `noqa` comments. But `pyproject.toml` will have to be edited manually.

Reply:
Agreed with @dhomeier -  I prefer not to be distracted by unnecessary `noqa`. So, we
1. Work around this rule with `from numpy import get_include as get_numpy_include`;
2. Ignore `setup_package.py`; or
3. Just make the few good changes here and simply keep ignoring `ICN001` - this is *not* a big deal one, and new contributions are very unlikely to "break" it.

Reply:
I'd be fine with `from numpy import get_include as get_numpy_include` or `from numpy import get_include as numpy_get_include`.

Reply:
I second your initial comment that creating some awkward custom imports merely to keep a linting tool happy or at least quiet feels rather silly; therefore I am not comfortable with the `get_numpy_include` option 1.
No. 2 would be my preferred solution – `pyproject.toml` already has 90+ lines of Ruff configuration with 20-something per-file ignores, so I don't see how another ignore could be considered unacceptable clutter. Same for 6 or 7 lines to clearly declare our own endorse aliases.
I probably could even live with `noqa` as next best option, rather than keeping the rule ignored (which would also require a new entry in `pyproject.toml`). We may not have needed it in any relevant code, and it is indeed unlikely a PR breaking it would be accepted, but it is just the purpose of an automated checker like this to ensure they really don't without anyone needing to think about it any more.